packages/queue/index.js

```diff
@@ -0,0 +1,103 @@
+const PgBoss = require('pg-boss');
+const log = require('@cardstack/logger')('cardstack/queue');
+
+module.exports = class Queue {
+  constructor() {
+    this.boss = null;
+    this.promiseResolveCallbacks = {};
+    this.promiseRejectCallbacks = {};
+    this.callbackedQueues = [];
+    this.jobErrors = {};
+  }
+
+  async _ensureBoss() {
+    if (!this.boss) {
+      this.boss = new PgBoss({ database: 'postgres', host: 'localhost', user: 'postgres', port: 5444 });
```

Comment:
My thoughts on your config question: 

 - this Queue class can take the whole `pgBossConfig` as a constructor argument
 - hub/queues.js can inject `config:pg-boss` and pass that into `new Queue`.
 - in hub/main.js `wireItUp`, would `registry.register('config:pg-boss', opts.pgBossConfig || { database: process.env.PG_BOSS_DB_NAME, ...etc })`

This allows the normal usage to be environment variables, while also allowing someone who is calling `createDefaultEnvironment` to pass the `pgBossConfig` option if they need to override.